scheduler/test/cook/test/scheduler/dru.clj

```diff
@@ -120,6 +122,45 @@
                       shuffle
                       (into {})))))))))
 
+;; This test makes sure that running and queued jobs are sorted with the right
+;; implcit order, in particular, it makes sure the codepath of create-task-ent
+;; created fake instances for waiting jobs sort in the right order with respect
+;; to actual running instances by task->feature-vector
+(deftest test-sorted-task-scored-task-pairs-with-running
+  (setup)
+  (let [datomic-uri "datomic:mem://test-sorted-task-scored-task-pairs-order"
+        conn (restore-fresh-database! datomic-uri)
+        job1 (create-dummy-job conn :user "ljin" :memory 10.0 :ncpus 10.0 :name "1")
+        job2 (create-dummy-job conn :user "ljin" :memory 20.0 :ncpus 20.0 :name "2" :job-state :job.state/running)
+        job3 (create-dummy-job conn :user "ljin" :memory 40.0 :ncpus 40.0 :name "3")
+        job4 (create-dummy-job conn :user "ljin" :memory 80.0 :ncpus 80.0 :name "4" :job-state :job.state/running)
+        job5 (create-dummy-job conn :user "ljin" :memory 160.0 :ncpus 160.0 :name "5")
```

Comment:
Should we mix the requests too? Like, should we avoid declaring them in sorted order?

Reply:
Oh interesting. We can file something for the backlog there.